simphony/io/h5_cuds.py

```diff
@@ -152,6 +155,47 @@ def add_mesh(self, mesh):
         self._file.flush()
         return m
 
+    def add_lattice(self, lattice, record=None):
```

Comment:
> We need a way of adding a lattice to a file with only a selected set of keywords, instead of all of them. This is currently done by using a custom record class that describes the Pytables columns.

What is not clear in this case is why you need to be able to define the record and what specific problem you are trying to solve. I thought that the data that are saved in a Lattice are a subset of CUBA keys that  is **fixed**. In that case there is no need to pass the `record` argument, one already knows what to use for a Lattice. However, if the list of keywords is not **fixed** then what is the advantage of having a smaller disk space footprint but every time one needs to add one more keywords, the complete Lattice instance has to be duplicated. In that case just use the full record.

To summarize, it looks that you are advocating that it is important to provide a record attribute for FileLattice in order to keep the disk space small based on the logic that this will speed up operations. The main problem is that changes like that should be first discussed with a usecase and a design proposal. This has happened in a number of small and large issues as one can find in github and in the wiki.

Note generic comment:

>2) Copying all the needed data from the original lattice to a new one, and adding that lattice to the H5CUDS-file. This gets rid of the record parameter in add_lattice, but it can multiply the memory requirements for the lattice (at the moment of creating the new one). Lattice-Boltzmann code is already run by filling the computer cluster node memory so every additional memory hungry operations should be avoided.

Independent of the fact that I like or not like this solution, it is not clear why one will need to make a full copy in memory of the Lattice when there are iterators that provide fast and memory efficient access to the containers.